### ISSUE TITLE
Add support selection to EOL commands

### ISSUE THREAD

**Issue by ArkadiuszMichalski** [notepad-plus-plus/notepad-plus-plus#12702]
Would be nice to have selection support for these two commands: `EOL to Space` and `Remove Unnecessary Blank and EOL`. Same as we have for trim operations https://github.com/notepad-plus-plus/notepad-plus-plus/pull/12655.

![image](https://user-images.githubusercontent.com/2730894/209535882-a5305430-da51-42a4-9f0e-d4a0e1b72ca7.png)

@donho I have a few questions before starting this. In `EOL` case it's a bit simpler, because here it's only important to select `EOL`, so what we select will be taken into account. So for `EOL to Space`:

1. 
![image](https://user-images.githubusercontent.com/2730894/209536658-6231a7f8-be96-4da2-9261-7a06e16224c7.png)
Will replace EOL in 1 line.

2. 
![image](https://user-images.githubusercontent.com/2730894/209536806-59f75121-58cc-49b0-aab4-7747001b161a.png)
Will replace EOL in 1 and 2 line.

3.
![image](https://user-images.githubusercontent.com/2730894/209536952-fd7a5039-5d1a-471f-a026-981f8040dcfe.png)
Will replace EOL in 1 line.

In this case it doesn't make sense to extend the selection to the whole data? For example:
![image](https://user-images.githubusercontent.com/2730894/209537144-5c9d031b-189a-48fe-92cb-e9365c3d355d.png)
Expanding will not affect the process because the selected EOL will not change. Well, but I'm not sure to not expand anyway, because the Trim operations make expanding and this one `EOL to Space` should not do it? Also `Remove Unnecessary Blank and EOL` firstly run `doTrim()` so expanding will be done here anyway.
These two commands should be considered in the context of whole lines or just according to the exactly selection made? A similar question will appear later for the last 3 commands `Tab to Space`, `Space to Tab (all/leading)`.

I have one more question about this command `Remove Unnecessary Blank and EOL`. Actually for such example:
![image](https://user-images.githubusercontent.com/2730894/209538580-c2d73df0-1192-459e-9615-889094df1709.png)
we get this:
![image](https://user-images.githubusercontent.com/2730894/209538128-1652e8e6-dad2-48a3-b924-14f7cb87ffac.png)

We get last 3 space because last 3 EOL was change to it. But this spaces are also not unnecessary, as command name suggest? Should not be removed?

One more example:
![image](https://user-images.githubusercontent.com/2730894/209540901-90e8b602-48d6-4f82-b415-dec8919ab3a3.png)
After all once space left. If we assume that for `Remove Unnecessary Blank and EOL`, actions for EOL always mean replacing them with a space, this might be correct.

**Comment by donho** [comment]
@ArkadiuszMichalski 
It makes sense that trim operations make expanding because trim operate for leading and trailing - that means the whole line.
Expanding is not necessary for replacing EOL, because it's different from blank operations (as you said: `Expanding will not affect the process because the selected EOL will not change`). We don't need to align all operations to have exactly the same behaviour. The most important thing to users (IMO) is, each operation makes sense and does what we expected.

**Comment by donho** [comment]
@ArkadiuszMichalski 

> I have one more question about this command Remove Unnecessary Blank and EOL. Actually for such example:

Oups! I have to admit that "Remove Unnecessary Blank and EOL" is not well named. This command trim leading & tail then replace EOL with space:
https://community.notepad-plus-plus.org/topic/17609/remove-unnecessary-blank-and-eol

> We get last 3 space because last 3 EOL was change to it. But this spaces are also not unnecessary, as command name suggest? Should not be removed?

Before changing its behaviour, I think it's worth to change the command name. What's your suggestion?

**Comment by alankilborn** [comment]
>worth to change the command name. What's your suggestion?

It is difficult.
"Join Lines Minimally" is fairly accurate, but clearly overlaps with the standard "Join Lines".

**Comment by ArkadiuszMichalski** [comment]
>We don't need to align all operations to have exactly the same behaviour. The most important thing to users (IMO) is, each operation makes sense and does what we expected.

OK, it's better to ask about it before making any changes.

>Before changing its behaviour, I think it's worth to change the command name. What's your suggestion?

`Trim both and EOL to Space` or `Trim all and EOL to Space`. Now it's just a sequence 3 and 4 commands so mention them directly would be less confusing.

**Comment by ArkadiuszMichalski** [comment]
I made some initial changes for the tests https://github.com/notepad-plus-plus/notepad-plus-plus/pull/12711.

Is `SCI_LINESJOIN` not always working correct?
![image](https://user-images.githubusercontent.com/2730894/209644747-e35de19b-01f8-46ae-8532-19876fd545e9.png)
This looks like a problem on Scintilla's side.

**Comment by ArkadiuszMichalski** [comment]
For above case:
https://sourceforge.net/p/scintilla/bugs/2372/
